pkg/apis/infinispan/v1/types_util.go

```diff
@@ -155,6 +155,9 @@ func (ispn *Infinispan) ApplyDefaults() {
 	if ispn.Spec.Security.EndpointAuthentication == nil {
 		ispn.Spec.Security.EndpointAuthentication = pointer.BoolPtr(true)
 	}
+	if *ispn.Spec.Security.EndpointAuthentication {
```

Comment:
@rigazilla please pay attention, that `endpointSecretName` never removed when changing `endpointAuthentication` from `true` to `false`